packages/dds/tree/src/core/rebase/fence.json

```diff
@@ -1,5 +1,5 @@
 {
 	"tags": ["rebase"],
 	"exports": ["index"],
-	"imports": ["id-compressor", "tree", "util"]
+	"imports": ["id-compressor", "tree", "util", "repair"]
```

Comment:
This introduces a cycle between repair and rebase, which would be nice to avoid.

I think the simplest way to avoid the cycle is to make RepairDataStore and ReadonlyRepairDataStore agnostic to the concrete revision tag type by making it a template parameter (you may even be able to give it the default type `unknown` if that helps reduce the number of changes in the code)).

Don't forget to update the diagram in the README to reflect the dependencies.